pkg/agent/plugin/nodeattestor/jointoken/join_token.go

```diff
@@ -18,25 +16,17 @@ func BuiltIn() catalog.BuiltIn {
 }
 
 func builtin(p *Plugin) catalog.BuiltIn {
-	return catalog.MakeBuiltIn(pluginName, nodeattestorv0.NodeAttestorPluginServer(p))
+	return catalog.MakeBuiltIn(pluginName, nodeattestorv1.NodeAttestorPluginServer(p))
 }
 
 type Plugin struct {
-	nodeattestorv0.UnsafeNodeAttestorServer
+	nodeattestorv1.UnsafeNodeAttestorServer
 }
 
 func New() *Plugin {
 	return &Plugin{}
 }
 
-func (p *Plugin) FetchAttestationData(stream nodeattestorv0.NodeAttestor_FetchAttestationDataServer) error {
-	return errors.New("join token attestation is currently implemented within the agent")
-}
-
-func (p *Plugin) Configure(ctx context.Context, req *spi.ConfigureRequest) (*spi.ConfigureResponse, error) {
-	return &spi.ConfigureResponse{}, nil
-}
-
-func (*Plugin) GetPluginInfo(context.Context, *spi.GetPluginInfoRequest) (*spi.GetPluginInfoResponse, error) {
-	return &spi.GetPluginInfoResponse{}, nil
+func (p *Plugin) AidAttestation(stream nodeattestorv1.NodeAttestor_AidAttestationServer) error {
+	return status.Error(codes.Internal, "join token attestation is currently implemented within the agent")
```

Comment:
I'm wondering if using `codes.Internal` is appropriate here. This looks more like an invalid argument / config rather than an internal failure of the plugin, so maybe `codes.InvalidArgument` fits better?

Reply:
I'm not sure it really matters what happens here. This plugin would only be called if we messed up somehow and didn't call the special join-token code during attestation, which is why I used `Internal`, since it implies we messed up internally.

Reply:
Isn't this reached when the join token plugin is present and a join token string is not specified? In that sense is that I see this more like an invalid argument rather than an internal misbehavior.

Reply:
Oh man. I think you are right. This whole message is completely wrong given that. Maybe we can merge this and I can open a specific PR to address that situation?

Reply:
Sure!

Reply:
:+1: